test/helpers.bash

```diff
@@ -318,6 +318,7 @@ function restart_crio() {
 }
 
 function cleanup_test() {
+	stop_crio
```

Comment:
If you add it here you can remove it from all the other places where it's currently called before `cleanup_test`, e.g. [here][1], with `cleanup_test` [here][2].  That may turn up a reason against this (e.g. a test that gets `cleanup_test` but no `start_crio`).

[1]: https://github.com/kubernetes-incubator/cri-o/blob/959afb79d4247891b6176b74609b1133cb939f1b/test/apparmor.bats#L36
[2]: https://github.com/kubernetes-incubator/cri-o/blob/959afb79d4247891b6176b74609b1133cb939f1b/test/apparmor.bats#L6

Reply:
Oh... interesting. Your first example shows what I think might be the crux of this issue: those cleanups at the end of a subtest will never get run if an intermediate step fails. My hunch is that all those `cleanup_xx` and `remove_apparmor_profile` steps should be moved into `teardown()`, possibly with safeguards added so they pass cleanly if unneeded (I haven't looked). Thanks for pointing this out.